### PR TITLE
hit: python bindings and TestHarness conversion

### LINKED ISSUE #5270: Andy: fix comments in your test-suite input files
Heather reports:

I have run into a problem where Peacock does something odd with some of the comments in the input file when it creates its own input file. For example, in initial_stress/gravity.i, this block:
  [./elasticity_tensor]

```
type = ComputeElasticityTensor

block = 0

fill_method = symmetric_isotropic

C_ijkl = '0.4 0.4' # young’s  = 1, poisson = 0.25
```

  [../]

Ends up like this:

  [./elasticity_tensor]

```
type = ComputeElasticityTensor

block = 0

fill_method = symmetric_isotropic

C_ijkl = '0.4 0.4' # young'

s = 1,

poisson = 0.25
```

  [../]

Which then obviously results in an error message when trying to run the simulation in Peacock. The problem goes away if I get rid of the equals signs from the comment.

### PR DESCRIPTION
**Pull request by rwcarlsen**
TestHarness uses the python bindings to parse tests files.  Now we get nice errors for tests files:

```
$ ./run_tests --re simple_diffusion
/Users/calsrw/moose/test/tests/kernels/simple_diffusion/tests:6: empty section name - did you mean '../'?
-------------------------------------------------------------------------------------------------------------
Ran 0 tests in 1.5 seconds
0 passed, 0 skipped, 0 pending, 0 failed, FATAL PARSER ERROR
```

This is still a bit rough and I'm not sure if this is exactly how we want to integrate it into the build system.  The cython-generated C code compiles with a bunch of warnings :-( - I'll probably end up disabling them for the bindings compilation.  I also still need to convert over other usages of the old python getpot parser in our python dir - shouldn't be too hard.

Fixes #5270, #7882, #8365

### CODE REVIEW

File: framework/contrib/hit/setup.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+cxxargs = ['-Wno-everything', '-std=c++11']
+
+# the -mmacosx-version-min flag is to get around the fact that distutils tries to compile c code
+# with the same flags that the python interpreter being used was compiled with.  For our conda
+# install, it was built with support for a minimum mac os x version of 10.6 - which makes it try
+# to use gcc's libstdc++ (which is old and doesn't support c++11) instead of clang's libc++ (which
+# properly supports c++11).  See
+# https://stackoverflow.com/questions/33738885/python-setuptools-not-including-c-standard-library-headers.
+if 'clang' in os.environ['CXX']:
+    cxxargs.append('-mmacosx-version-min=10.9')
```
Comment (permcody) [inline framework/contrib/hit/setup.py]: Does this flag cause an issue on Linux?

File: framework/moose.mk [hunk 2]
```diff
@@ -28,6 +28,19 @@ hit_LIB       := $(hit_DIR)/libhit-$(METHOD).la
 hit_deps      := $(patsubst %.cc, %.$(obj-suffix).d, $(hit_srcfiles))
 
 #
+# hit python bindings
+#
+pyhit_srcfiles  := $(hit_DIR)/hit.cpp $(hit_DIR)/lex.cc $(hit_DIR)/parse.cc
+pyhit_LIB       := $(hit_DIR)/hit.so
+
+# the sed hack here is to get around a bug in python's distutils that only uses the first word of
```
Comment (permcody) [inline framework/moose.mk]: What sed hack?

**Review by permcody — CHANGES_REQUESTED** [review]
This looks fine to me. We just need a few instructions for maintainers now. When/why/how we regenerate these bindings in a README file or some other place we can easily locate it. A separate make target would be awesome.

Finally, you might want to start documenting your HIT parser a little bit more. Right now, I'm pretty certain only you could maintain it. It's a sparse as they come in terms of code comments.

File: framework/contrib/hit/setup.py (hunk 1 above)
Comment (rwcarlsen) [reply framework/contrib/hit/setup.py]: gcc complains that it doesn't know what the flag means.

File: framework/moose.mk (hunk 2 above)
Comment (rwcarlsen) [reply framework/moose.mk]: I removed the hack - I'll remove the comment.

File: framework/contrib/hit/setup.py (hunk 1 above)
Comment (permcody) [reply framework/contrib/hit/setup.py]: Well GCC and Clang both work on Mac and Linux. The compiler is independent of the platform. I'm wondering if we need to look at the platform AND compiler to add this flag.

**Comment by rwcarlsen** [comment]
The parser has over 200 lines of comments for ~1400 lines of code!  Are there particular parts or something specific you think needs more comment attention (e.g. high-level structure or implementation trick details)?

**Comment by rwcarlsen** [comment]
You are right - we should check the OS too.

**Comment by permcody** [comment]
On comments, I was looking at the new pyx file. It's sparse. I recall the original parser being sparse but I could be recalling incorrectly.

File: framework/contrib/hit/setup.py (hunk 1 above)
Comment (rwcarlsen) [reply framework/contrib/hit/setup.py]: You are right - we should check the OS too.

**Comment by rwcarlsen** [comment]
The pyx file mostly mirrors the structure and functionality of the c++ code so I was perhaps more lazy than I should be - since there are some important differences such as parameter type handling.  I'll give it some doc love.

**Comment by rwcarlsen** [comment]
Green green green green green.  There is a relap-7 PR submitted that will need to go in as soon as this hits moose master.

**Review by permcody — APPROVED** [review]

**Pull request merged**